### PR TITLE
add static and shared Library for analysis and IR

### PR DESCRIPTION
**Pull request by luotao1**
This PR is the first version of static and shared library including analysis and IR.
In the next PRs, will do:
- add analysis and IR demo_CI

### CODE REVIEW

File: paddle/fluid/framework/ir/CMakeLists.txt [hunk 1]
```diff
@@ -18,3 +18,4 @@ cc_test(graph_helper_test SRCS graph_helper_test.cc DEPS graph graph_helper op_r
 cc_test(graph_to_program_pass_test SRCS graph_to_program_pass_test.cc DEPS graph_to_program_pass)
 cc_test(test_graph_pattern_detector SRCS graph_pattern_detector_tester.cc DEPS graph_pattern_detector)
 cc_test(test_fc_fuse_pass SRCS fc_fuse_pass_tester.cc DEPS fc_fuse_pass graph_pattern_detector graph pass graph_traits framework_proto)
+set(GLOB_PASS_LIB fc_fuse_pass attention_lstm_fuse_pass infer_clean_graph_pass fc_lstm_fuse_pass seq_concat_fc_fuse_pass)
```
Comment (luotao1) [inline paddle/fluid/framework/ir/CMakeLists.txt]: In next PRs, will auto update `GLOB_PASS_LIB` like `GLOB_OP_LIB` in `operators/CMakeLists.txt`

File: paddle/fluid/inference/api/analysis_predictor.cc [hunk 2]
```diff
@@ -135,5 +135,8 @@ std::unique_ptr<PaddlePredictor> CreatePaddlePredictor<
 }  // namespace paddle
 
 USE_PASS(fc_fuse_pass);
+USE_PASS(seq_concat_fc_fuse_pass);
+USE_PASS(fc_lstm_fuse_pass);
 USE_PASS(graph_viz_pass);
 USE_PASS(infer_clean_graph_pass);
+USE_PASS(attention_lstm_fuse_pass);
```
Comment (luotao1) [inline paddle/fluid/inference/api/analysis_predictor.cc]: In the next PRs, will auto-generate a `pass.h` like `pybind.h` including all `USE_OP(XXX)`.

File: paddle/fluid/framework/ir/CMakeLists.txt (hunk 1 above)
Comment (luotao1) [reply paddle/fluid/framework/ir/CMakeLists.txt]: auto update is finished in https://github.com/PaddlePaddle/Paddle/pull/13133/commits/d7b496578559e18265bd91d8dc5a7b46c80ae997

File: paddle/fluid/inference/api/analysis_predictor.cc (hunk 2 above)
Comment (luotao1) [reply paddle/fluid/inference/api/analysis_predictor.cc]: auto-generate is finished in https://github.com/PaddlePaddle/Paddle/pull/13133/commits/d7b496578559e18265bd91d8dc5a7b46c80ae997

**Review by Superjomn — APPROVED** [review]
LGTM

**Pull request merged**